tests/test_neighbors.py

```diff
@@ -0,0 +1,78 @@
+from gamelib.BoardItem import BoardItem, BoardItemVoid
+from gamelib.Game import Game
+from gamelib.Characters import Player
+import gamelib.Constants as Constants
+import unittest
+
+
+def readable_board_item(boarditem):
+    return f"{boarditem.pos}\t{boarditem.name.strip()}"
+
+
+def readable_board_items(board_item_list):
+    return "[" + ", ".join([readable_board_item(item) for item in board_item_list]) + "]"
+
+
+class GameNeighborTestCase(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.game = Game()
+        self.board = self.game.load_board("hac-maps/kneighbors.json", 1)
+        self.game.player = Player(name='player')
+        self.game.change_level(1)
+        self.tree26 = self.board.item(2, 6)
+        self.treasure38 = self.board.item(3, 8)
+        self.tree39 = self.board.item(3, 9)
+        self.wall45 = self.board.item(4, 5)
+        self.wall46 = self.board.item(4, 6)
+        self.wall47 = self.board.item(4, 7)
+        self.wall57 = self.board.item(5, 7)
+        self.npc77 = self.board.item(7, 7)
+        self.treasure1212 = self.board.item(12, 12)
+        self.tree1310 = self.board.item(13, 10)
+        self.npc168 = self.board.item(16, 8)
+
+    def test_player_neighbors(self):
+        testcases = (
+            ([3, 3], []),
+            ([3, 5], [self.tree26, self.wall45, self.wall46]),
+            ([3, 6], [self.tree26, self.wall45, self.wall46, self.wall47]),
+            ([3, 7], [self.tree26, self.wall46, self.wall47, self.treasure38]),
+            ([4, 8], [self.wall47, self.wall57, self.treasure38, self.tree39]),
+            ([5, 6], [self.wall45, self.wall46, self.wall47, self.wall57]),
+        )
+
+        for pos, expected_board_items in testcases:
+            with self.subTest(str(pos)):
+                self.game.player.pos = pos
+                actual_board_items = self.game.neighbors()
+                assertion_message = " ".join([
+                    readable_board_items(expected_board_items),
+                    "!=",
+                    readable_board_items(actual_board_items)
+                ])
+                self.assertSetEqual(set(expected_board_items), set(actual_board_items), assertion_message)
+
+    def test_npc_neighbors(self):
+        def move_player_next_to_npc77():
+            pos_next_to_npc = [7, 8]
+            self.game.move_player(Constants.DOWN, pos_next_to_npc[0])
```

Comment:
@arnauddupuis Yes, I totally meant to use that second parameter and was glad that is was there =). Although I have to admit I was not sure what is the best way to move the player next to the NPC, that's why I created this method so that the intention is clear. Perhaps a direct teleportation to a given coordinate can be accomplished more easily, but I was not sure about possible side effects (what exactly gets updated and when, to be precise), so the normal move_player seemed to be the safest option.

Reply:
It definitely is, thanks a lot I'll merge your PR right away.